document/chapters/preliminaries.tex

```diff
@@ -1,3 +1,190 @@
 % !TEX root = ../main.tex
 \chapter{Preliminaries}
 \label{chapter:preliminaries}
+
+Rust is a systems programming language focused on speed, memory safety and concurrency. It intends to offer both performance similar to C++ and memory safety similar to Haskell. Rust is a compiled language having imperative and functional programming features. Its performance makes Rust an ideal candidate for writing operative systems, databases, compilers and other high-performance software without worrying about manual memory allocation. However, its safety and high-level abstractions has encouraged its usage in backend and even frontend development. 
```

Comment:
Rust ```Add a reference to Rust (if its the URL to the webpage as a footnote, if it is a paper, as a references``` is a systems programming language focused on speed, memory safety and concurrency. It intends to offer both performance similar to C++ and memory safety similar to Haskell. Rust is a compiled language ```combining``` imperative and functional programming features. ~~Its~~```The focus on ``` performance makes Rust an ideal candidate for writing operative systems, databases, compilers```, (oxford comma)``` and other high-performance software without worrying about manual memory allocation. However, ~~its~~ ```Refer to the concrete Object``` ```Rust's```safety and high-level abstractions has encouraged its usage in backend and even frontend development.

Reply:
Solved on https://github.com/FLAGlab/bounded-const-generics-rust/pull/6/commits/57518d1ac39622379e8318b8aab2f2ac24be5a9a